### PR TITLE
[QoS] Implement QoS checks for Solana v0

### PR DESCRIPTION
**Pull request by adshmh**
This PR implements the first version of QoS functionality for the Solana blockchain service endpoints.
It follows the general pattern established for the `qos` package, i.e.:
- Build a QoS-specific request context from a request
- Parse Solana-specific responses in the established context.
- Generate QoS checks for Solana using the new response parsers.

### CODE REVIEW

File: qos/solana/context.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	}
+
+	return observationSet{
+		EndpointStore: rc.EndpointStore,
+		ServiceState:  rc.ServiceState,
+		Observations:  observations,
+	}
+}
+
+func (rc *requestContext) GetEndpointSelector() relayer.EndpointSelector {
```
Comment (commoddity) [inline qos/solana/context.go:164]: Can you add a comment describing that the `requestContext` satisfies the `relayer.EndpointSelector` interface and why that is.

File: qos/solana/endpoint.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	// TODO_FUTURE: support archival endpoints.
+}
+
+// ValidateBasic returns an error if the endpoint is invalid regardless of the state of the service.
+// e.g. an endpoint without an observation of its response to a `GetHealth` request is not considered valid.
+func (e endpoint) ValidateBasic() error {
+	switch {
+	case e.GetHealthResult == nil:
+		return fmt.Errorf("endpoint has not had an observation of its response to a %q request", "getHealth")
```
Comment (commoddity) [inline qos/solana/endpoint.go:34]: WDYT about storing the error strings as a group of named vars at the top of the file?

File: qos/solana/observe.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+// of QoS data between PATH instances, and the updating of local
+// PATH instance's QoS data on Solana endpoints.
+var _ message.ObservationSet = observationSet{}
+
+// observation captures the result of processing an endpoint's
+// response to a service request.
+// It provides details needed to establish the validity of
+// an endpoint.
+type observation interface {
+	// Apply updates the endpoint based on the observation's contents.
```
Comment (commoddity) [inline qos/solana/observe.go:23]: This is great, love the simplicity of providing the `Apply` method for updating endpoint results.

**Review by commoddity — CHANGES_REQUESTED** [review]
Looks very good on initial review. I am liking the design of the new QoS framework; very clear and easy to follow. Only a couple minor comments.

File: qos/solana/endpoint.go (hunk 2 above)
Comment (adshmh) [reply qos/solana/endpoint.go:34]: Thanks for catching this. Fixed.

**Pull request merged**